src/mapnik_image_encode.cpp

```diff
@@ -205,7 +216,8 @@ Napi::Value Image::encode(Napi::CallbackInfo const& info)
         return env.Undefined();
     }
     Napi::Function callback = callback_val.As<Napi::Function>();
-    auto* worker = new AsyncEncode{image_, palette, format, callback};
+    this->Ref();
```

Comment:
Why, in this case, do you `Ref()` here rather than inside the `AsyncEncode` ctor? Asking because I presume the clearest RAII pattern would be to use matching ctor and dtor for Ref/Unref.

Reply:
In this case `Image` object gets `Unref()` in 'OnWorkComplete` so RAII is not really helpful. I found this way `Image` is kept alive for shorter periods as we don't have control over `AsyncEncode` object lifetime. I also experienced crashes when calling `Unref()` in dtor which most likely not related to this particular issue. I'll run some tests and  get back on this one. Thanks for spotting unusual usage :) !